project/SparkRedshiftBuild.scala

```diff
@@ -52,6 +52,7 @@ object SparkRedshiftBuild extends Build {
         "Spark 1.5.0 RC2 Staging" at "https://repository.apache.org/content/repositories/orgapachespark-1141",
       libraryDependencies ++= Seq(
         "com.amazonaws" % "aws-java-sdk-core" % "1.9.40" % "provided",
+        "com.amazonaws" % "aws-java-sdk-s3" % "1.9.40" % "provided",
```

Comment:
To try to reason through these points:

- I don't think that marking these libraries as required dependencies should be an issue on EMR unless Spark's `userClasspathFirst` setting is enabled, at least in the sense that `spark-redshift`'s dependencies on these libraries should not cause the Amazon-provided versions to be evicted / overridden on the classpath.
- Taking Databricks Cloud as an example of a non-EMR system which interacts with S3, I believe that we also provide `hadoop-aws` and the AWS SDK.

I guess the real concern here is not that we necessarily override the environment's copy of the AWS SDK, but that the SDK's dependency on Joda, Jackson, and Guava ends up pulling in newer versions of those libraries which evict earlier versions that some of the rest of the user-provided libraries might rely on. However, I also wonder if those same users might run into trouble anyways if the environment-provided version of the library causes its versions of Joda, Jackson, and Guava to take precedence when `userClasspathFirst=false`.

In a nutshell, I think this is a decision that could easily go either way. Based on discussion above, I don't think that it's necessarily going to make a difference for Databricks or EMR users; I think the impact would be most heavily concentrated on users with version-conflict-prone dependencies on the SDK's dependencies who deploy with `userClasspathFirst` or deploy in environments where this dependency is not provided.